consul/tests/conftest.py

```diff
@@ -1,74 +1,71 @@
-# (C) Datadog, Inc. 2010-2017
+# (C) Datadog, Inc. 2018
 # All rights reserved
-# Licensed under Simplified BSD License (see LICENSE)
-import subprocess
+# Licensed under a 3-clause BSD style license (see LICENSE)
 import os
-import logging
 
 import pytest
 import requests
 
-import common
+from datadog_checks.dev import WaitFor, docker_run
+from . import common
 
-log = logging.getLogger(__file__)
 
+def _consul_config_path():
+    server_file = 'server-{}.json'.format(os.getenv('CONSUL_VERSION'))
+    return os.path.join(common.HERE, 'compose', server_file)
 
-def wait_for_cluster():
+
+def ping_cluster():
     """
     Wait for the slave to connect to the master
     """
-    from time import sleep
-    for i in xrange(0, 50):
-        sleep(1)
-        try:
-            res = requests.get("{}/v1/status/peers".format(common.URL))
-            # Wait for all 3 agents to join the cluster
-            if len(res.json()) == 3:
-                return True
-        except Exception as e:
-            log.info("Error connecting to the cluster: %s", e)
-            pass
+    response = requests.get('{}/v1/status/peers'.format(common.URL))
+    response.raise_for_status()
+
+    # Wait for all 3 agents to join the cluster
+    if len(response.json()) == 3:
+        return True
 
     return False
 
 
-@pytest.fixture(scope="session")
-def consul_cluster():
+@pytest.fixture(scope='session')
+def dd_environment(instance):
```

Comment:
I think if we put the `scope=module` and `autouse=True` then we wouldn't have to put this fixture as an argument to each function. Though that'd require making unit and integration tests their own modules, which may be slightly more complex than we want. What do you think?

Reply:
I like